crates/ruff_python_formatter/src/comments/node_key.rs

```diff
@@ -0,0 +1,169 @@
+use ruff_python_ast::node::AnyNodeRef;
+use std::fmt::{Debug, Formatter};
+use std::hash::{Hash, Hasher};
+use std::ptr::NonNull;
+
+#[derive(Copy, Clone)]
+pub(super) struct NodeRefEqualityKey<'a> {
```

Comment:
Probably need to document where this is used for, I got lost at this point :-)

Is this just a pointer?

Reply:
It's used as keys into the `Comments` struct because `AnyNodeRef` does not implement `Hash` (and we want hashing to be cheap.)